### DISCUSSION TITLE
Coverage of UI tests (Selenium, ASP.NET Core)

### DISCUSSION THREAD

**Discussion by Piedone** [SteveGilham/altcover#136] [Q&A] [answered]
I'd like to collect coverage data for Selenium UI tests of an ASP.NET Core web app. The tests are executed with `dotnet test`, which run the app with the `dotnet` CLI behind the scenes. Does altcover support this scenario? 

It seems to me that is does with the [Instrument now, test later, collect coverage after that](https://github.com/SteveGilham/altcover/wiki/Modes-of-Operation#instrument-now-test-later-collect-coverage-after-that) mode. In my understanding, instrumented apps can then be executed in any way.

Thank you!

**Comment by SteveGilham** [comment, marked as answer]
Yes, that would be the way to do it -- and is in fact how the simple `dotnet test` integration for unit tests already works under the covers, by executing a Prepare step ahead of invoking the VsTest step for test execution, and a Collect step after, using the MSBuild tasks exposed by the AltCover.Engine assembly.

**Reply by Piedone** [reply]
Great, thank you!
